cmd/clusterctl/client/repository/repository_memory_test.go

```diff
@@ -0,0 +1,156 @@
+				mr.WithPaths("", "components.yaml")
+				mr.WithDefaultVersion("v2.0.0")
+			},
+			want: want{
+				versions:       []string{"v1.0.0", "v2.0.0"},
+				defaultVersion: []byte("v2.0.0"),
+				latestVersion:  []byte("v2.0.0"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "GetFile can use latest as default",
+			addTestData: func(mr *MemoryRepository) {
+				mr.WithFile("v1.0.0", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v1.0.0", "components.yaml", []byte("v1.0.0"))
+				mr.WithFile("v2.0.0", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v2.0.0", "components.yaml", []byte("v2.0.0"))
+				mr.WithPaths("", "components.yaml")
+				mr.WithDefaultVersion("latest")
+			},
+			want: want{
+				versions:       []string{"v1.0.0", "v2.0.0"},
+				defaultVersion: []byte("v2.0.0"),
+				latestVersion:  []byte("v2.0.0"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Get all valid releases available from release directory",
+			addTestData: func(mr *MemoryRepository) {
+				mr.WithFile("v1.0.0", "components.yaml", []byte("v1.0.0"))
+				mr.WithFile("v1.0.0", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v1.0.1", "components.yaml", []byte("v1.0.1"))
+				mr.WithFile("v1.0.1", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v2.0.1", "components.yaml", []byte("v2.0.1"))
+				mr.WithFile("v2.0.1", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v2.0.2+exp.sha.5114f85", "components.yaml", []byte("v2.0.2+exp.sha.5114f85"))
+				mr.WithFile("v2.0.2+exp.sha.5114f85", "metadata.yaml", []byte(metadata))
+				mr.WithFile("v2.0.3-alpha", "components.yaml", []byte("v2.0.3-alpha"))
+				mr.WithFile("v2.0.3-alpha", "metadata.yaml", []byte(metadata))
+				mr.WithPaths("", "components.yaml")
+			},
+			want: want{
+				versions:       []string{"v1.0.0", "v1.0.1", "v2.0.1", "v2.0.2+exp.sha.5114f85", "v2.0.3-alpha"},
+				defaultVersion: []byte("v1.0.0"),
+				latestVersion:  []byte("v2.0.2+exp.sha.5114f85"),
+			},
+			wantErr: false,
+		},
+		{
+			name: "Get pre-release",
+			addTestData: func(mr *MemoryRepository) {
+				mr.WithFile("v2.0.3-alpha", "components.yaml", []byte("v2.0.3-alpha"))
+				mr.WithFile("v2.0.3-alpha", "metadata.yaml", []byte(metadata))
+				mr.WithPaths("", "components.yaml")
+			},
+			want: want{
+				versions:       []string{"v2.0.3-alpha"},
+				defaultVersion: []byte("v2.0.3-alpha"),
+				latestVersion:  []byte("v2.0.3-alpha"),
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+
+			r := NewMemoryRepository()
+			tt.addTestData(r)
+
+			g.Expect(r.RootPath()).To(Equal(""))
+
+			g.Expect(r.GetFile(r.DefaultVersion(), r.ComponentsPath())).To(Equal(tt.want.defaultVersion))
+			g.Expect(r.GetFile("", r.ComponentsPath())).To(Equal(tt.want.defaultVersion))
+			g.Expect(r.GetFile("latest", r.ComponentsPath())).To(Equal(tt.want.latestVersion))
+
+			got, err := r.GetVersions()
+			if tt.wantErr {
```

Comment:
I think it would be good to either rename wantErr to make clear it's related to GetVersions or just always expect that we didn't get an error and drop the field (currently it's always false)